src/NHibernate.Test/SystemTransactions/SystemTransactionFixtureBase.cs

```diff
@@ -39,8 +39,12 @@ protected void DisableConnectionAutoEnlist(Configuration configuration)
 			}
 			// Purge any previous enlist
 			connectionString = Regex.Replace(
-				connectionString, $"[^;\"a-zA-Z]*{autoEnlistmentKeywordPattern}=[^;\"]*", string.Empty,
-				RegexOptions.IgnoreCase | RegexOptions.Multiline);
+				connectionString, $"[^;\"a-zA-Z]*{autoEnlistmentKeywordPattern}=[^;\"]*;?", string.Empty,
+				RegexOptions.IgnoreCase);
+			// Avoid redundant semi-colon
+			connectionString = Regex.Replace(
```

Comment:
`HanaConnection` does not support them, so I had to improve this connection string patching.